src/credentials/CHIPCert.h

```diff
@@ -0,0 +1,460 @@
+    // script.
+    //
+    // WARNING! Assign no values higher than 127.
+    //
+    kTag_DNAttrType_CommonName              = 1,  /**< [ UTF8 string ] */
+    kTag_DNAttrType_Surname                 = 2,  /**< [ UTF8 string ] */
+    kTag_DNAttrType_SerialNumber            = 3,  /**< [ UTF8 string ] */
+    kTag_DNAttrType_CountryName             = 4,  /**< [ UTF8 string ] */
+    kTag_DNAttrType_LocalityName            = 5,  /**< [ UTF8 string ] */
+    kTag_DNAttrType_StateOrProvinceName     = 6,  /**< [ UTF8 string ] */
+    kTag_DNAttrType_OrganizationName        = 7,  /**< [ UTF8 string ] */
+    kTag_DNAttrType_OrganizationalUnitName  = 8,  /**< [ UTF8 string ] */
+    kTag_DNAttrType_Title                   = 9,  /**< [ UTF8 string ] */
+    kTag_DNAttrType_Name                    = 10, /**< [ UTF8 string ] */
+    kTag_DNAttrType_GivenName               = 11, /**< [ UTF8 string ] */
+    kTag_DNAttrType_Initials                = 12, /**< [ UTF8 string ] */
+    kTag_DNAttrType_GenerationQualifier     = 13, /**< [ UTF8 string ] */
+    kTag_DNAttrType_DNQualifier             = 14, /**< [ UTF8 string ] */
+    kTag_DNAttrType_Pseudonym               = 15, /**< [ UTF8 string ] */
+    kTag_DNAttrType_DomainComponent         = 16, /**< [ UTF8 string ] */
+    kTag_DNAttrType_ChipDeviceId            = 17, /**< [ unsigned int ] */
+    kTag_DNAttrType_ChipServiceEndpointId   = 18, /**< [ unsigned int ] */
+    kTag_DNAttrType_ChipCAId                = 19, /**< [ unsigned int ] */
+    kTag_DNAttrType_ChipSoftwarePublisherId = 20, /**< [ unsigned int ] */
+};
+
+/** Identifies the purpose or application of certificate
+ *
+ * A certificate type is a label that describes a certificate's purpose or application.
+ * Certificate types are not carried as attributes of the corresponding certificates, but
+ * rather are derived from the certificate's structure and/or the context in which it is used.
+ * The certificate type enumeration includes a set of pre-defined values describing commonly
+ * used certificate applications.  Developers can also extend the certificate type value
+ * range with application-specific types that described custom certificates or certificates
+ * with unique security properties.
+ *
+ * Certificate types are primarily used in the implementation of access control policies,
+ * where access to application features is influenced by the type of certificate presented
+ * by a requester.
+ *
+ * @note Cert type is an API data type only; it should never be sent over-the-wire.
+ */
+enum
+{
+    kCertType_NotSpecified    = 0x00, /**< The certificate's type has not been specified. */
+    kCertType_General         = 0x01, /**< The certificate is of a general or non-specific type. */
+    kCertType_Node            = 0x02, /**< A CHIP node certificate. */
+    kCertType_FirmwareSigning = 0x03, /**< A CHIP firmware signing certificate. */
```

Comment:
I've not been involved in the OTA update discussions but will OTA updates be (potentially) signed by a CHIP certificate ?